util/test/Test/Pos/Util/Tripping.hs

```diff
@@ -1,34 +1,52 @@
-{-# OPTIONS_GHC -fno-warn-orphans #-}
+{-# OPTIONS_GHC -fno-warn-orphans      #-}
+{-# OPTIONS_GHC -fno-warn-deprecations #-}
 
 module Test.Pos.Util.Tripping where
 
 import qualified Prelude
 import           Universum
 
-import           Data.Aeson (FromJSON, ToJSON, decode, encode)
+import           Data.Aeson (FromJSON, ToJSON, eitherDecode, encode)
 import           Data.Text.Internal.Builder (fromText, toLazyText)
+import qualified Data.Yaml as Y
 import           Formatting.Buildable (Buildable (..))
-import           Hedgehog (Group, MonadTest, discoverPrefix, success, tripping)
-import           Hedgehog.Internal.Property (Diff (..), failWith)
-import           Hedgehog.Internal.Show (valueDiff)
+import           Hedgehog (Gen, Group, MonadTest, discoverPrefix, success, tripping)
+import           Hedgehog.Internal.Property (Diff (..), Property, TestLimit, annotate,
+                     failWith, forAllWith, property, withTests)
+import           Hedgehog.Internal.Show (showPretty, valueDiff)
 import           Hedgehog.Internal.TH (TExpQ)
 import           System.IO (hSetEncoding, stderr, stdout, utf8)
 import qualified Text.JSON.Canonical as Canonical
 import           Text.Show.Pretty (Value (..), parseValue)
 
 import           Pos.Util.Json.Canonical (SchemaError (..))
 
+roundTripsAesonYamlShow :: (Eq a, Show a, ToJSON a, FromJSON a) => TestLimit -> Gen a -> Property
+roundTripsAesonYamlShow testLimit things = withTests testLimit . property $ do
+     annotate "Aeson"
+     forAllWith
+         (\x -> yamlAesonCustomRender x encode eitherDecode) things >>= roundTripsAesonShow
```

Comment:
`yamlAesonCustomRender` was necessary because for some reason, `Hedgehog` would not render the diffs nicely (and only return the value that failed the tests) when `tripping` was wrapped (`eachOf``: https://github.com/input-output-hk/cardano-sl/blob/bcacaa9b5a78f48c7db347f73fec2fe5421e0dfd/util/test/Test/Pos/Util/Golden.hs#L31)